### PR TITLE
Makefile: prefer to use podman if available as a container engine

### PR DESCRIPTION
**Pull request by mykaul**
in Makefile.

If podman is available, use it. Otherwise, use docker commands
in the Makefile

### Explain the changes
1. 

### Issues: Fixed #xxx / Gap #xxx
1. 

### Testing Instructions:
1.

### CODE REVIEW

**Review by liranmauda — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -1,5 +1,6 @@
 GIT_COMMIT?="$(shell git rev-parse HEAD | head -c 7)"
-NAME_POSTFIX?="$(shell docker ps -a | wc -l | xargs)"
+CONTAINER_ENGINE?=$(shell podman version >/dev/null 2>&1 && echo podman || echo docker)
```
Comment (nimrod-becker) [inline Makefile]:
```suggestion
CONTAINER_ENGINE?="docker"
```
Let's have it as an outside parameter for cases in which docker is not available (it  is available on Linux), don't think we want an automatic (and default) behaviour which can change the engine we build with.

Comment (mykaul) [reply Makefile]: The whole point was to prefer podman over docker, if available.
If it doesn't make sense, I can drop this.
The reason I needed this was that Docker is disabled on my Fedora as I'm using podman (without root privs., etc.).

Comment (nimrod-becker) [reply Makefile]: I think that supporting podman is good. I don't think it should be preferred.
So adding a parameter to use it so people without docker would be able to build is good.

**Comment by nimrod-becker** [comment]
Fixed in #5887

**Pull request closed without merging**